x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/InternalTopMetrics.java

```diff
@@ -169,9 +169,31 @@ public double value(String name) {
         if (topMetrics.isEmpty()) {
             return Double.NaN;
         }
-        assert topMetrics.size() == 1 : "property paths should only resolve against top metrics with size == 1.";
+
+        MetricValue value = topMetrics.get(0).metricValues.get(index);
+        if (value == null) {
+            return Double.NaN;
+        }
+        return value.numberValue().doubleValue();
+
+        //assert topMetrics.size() == 1 : "property paths should only resolve against top metrics with size == 1.";
         // TODO it'd probably be nicer to have "compareTo" instead of assuming a double.
-        return topMetrics.get(0).metricValues.get(index).numberValue().doubleValue();
+        //return topMetrics.get(0).metricValues.get(index).numberValue().doubleValue();
+    }
+
+    @Override
+    public String valueAsString(String name) {
+        int index = metricNames.indexOf(name);
+        if (index < 0) {
+            throw new IllegalArgumentException("unknown metric [" + name + "]");
+        }
+        if (topMetrics.isEmpty()) {
+            return "";
+        }
+        assert topMetrics.size() == 1 : "property paths should only resolve against top metrics with size == 1.";
+
+        MetricValue value = topMetrics.get(0).metricValues.get(index);
+        return value.getValue().format(value.getFormat());
```

Comment:
^ @nik9000 This is what I need to override. Of course this will work _without_ the change in `InternalNumericMetricsAggregation.java` (the former default would take the double and call to string on it)